### PR TITLE
feat(cli): Improve CLI logging

### PR DESCRIPTION
**Pull request by JonasKruckenberg**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [ ] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### Other information

This PR overhauls the CLI's output, completely moving it from a custom logging solution to the standard `log` and `env_logger` crates. Towards this goal I removed **all** printing functionality from the bundler, cleaning up concerns and bringing the bundler one step closer to being a reusable library.

### Major new feature

- The CLI now offers *way* more debug output, logging parameters and outputs of each executed command and shell script plus more. 
- The produced output now look much closer to cargo's output (and IMO better than before)
      For this we leverage a custom KV attribute for log macros: `action` that will take the log levels position in the output. `info!(action = "Bundling"; "{}", app_product_name);` produces `Bundling <App Name>`
- The log level can be controlled via the `--verbose` flag and the `RUST_LOG` env var. (The flag can appear multiple times, where `--verbose` means `Debug` and `--verbose --verbose` means `Trace`)

### CODE REVIEW

**Comment by JonasKruckenberg** [comment]
I think this doesn't compile on linux or windows 🤔 but my rust-analyzer seems to be broken rn

File: tooling/bundler/src/bundle/settings.rs [hunk 1]
```diff
@@ -386,7 +384,6 @@ pub struct Settings {
 #[derive(Default)]
 pub struct SettingsBuilder {
   project_out_directory: Option<PathBuf>,
-  verbose: bool,
```
Comment (lucasfernog) [inline tooling/bundler/src/bundle/settings.rs:389]: we should mark this change as a breaking change in the change file, and show the solution (i know there's some people using the tauri bundler as a library)

**Review by lucasfernog — CHANGES_REQUESTED** [review]
top5 best Tauri PRs easily

**Comment by amrbashir** [comment]
Bug alert, on Windows, if you have `before(Build|Dev)Command`, it prints the `before(Build|Dev)Command` output then hangs indefinitely and if `before(Build|Dev)Command` clears the terminal like how vite devServer does by default, it will only print `Running BeforeDevCommand (`pnpm dev`)`.

File: tooling/bundler/src/bundle/settings.rs (hunk 1 above)
Comment (JonasKruckenberg) [reply tooling/bundler/src/bundle/settings.rs:389]: True! Good catch

**Comment by JonasKruckenberg** [comment]
> Bug alert, on Windows, if you have `before(Build|Dev)Command`, it prints the `before(Build|Dev)Command` output then hangs indefinitely and if `before(Build|Dev)Command` clears the terminal like how vite devServer does by default, it will only print `Running BeforeDevCommand (`pnpm dev`)`. 

Is that using the latest commit from @lucasfernog? Maybe it has to do with the piping? I had a bunch of issues while developing with disappearing output too.

Maybe not though, since it's the same implementation as before basically.

@lucasfernog do we need `os_pipe` in the first place? We could just use the standard `Stdio::inherit()` if don't need the output.

**Comment by lucasfernog** [comment]
We do need the os_pipe @JonasKruckenberg otherwise the binding just won't show the output.

**Comment by lucasfernog** [comment]
@amrbashir that was fixed in my commits.

**Comment by amrbashir** [comment]
Oh then I might've checked with an older commit. I tested the branch twice and I missed a git pull in between.

Edit: just tested latest commits and it works.

**Review by lucasfernog — APPROVED** [review]

**Comment by JonasKruckenberg** [comment]
Niiice 🥳

**Pull request merged**